src/pharmpy/tools/run_amd.py

```diff
@@ -126,48 +126,64 @@ def run_amd(
 
     db = default_tool_database(toolname='amd')
     fit(model)
-    run_subfuncs = dict()
+    run_tool('modelfit', model, path=db.path / 'modelfit')
+    next_model = model
+    run_subtools = dict()
     for section in order:
         if section == 'structural':
             func = partial(_run_modelsearch, search_space=search_space, path=db.path)
-            run_subfuncs['modelsearch'] = func
+            subresults = func(next_model)
+            next_model = subresults.best_model
+            run_subtools['modelsearch'] = subresults
         elif section == 'iivsearch':
             func = partial(_run_iiv, path=db.path)
-            run_subfuncs['iivsearch'] = func
+            subresults = func(next_model)
+            next_model = subresults.best_model
+            run_subtools['iivsearch'] = subresults
         elif section == 'iovsearch':
             func = partial(_run_iov, occasion=occasion, path=db.path)
-            run_subfuncs['iovsearch'] = func
+            subresults = func(next_model)
+            next_model = subresults.best_model
+            run_subtools['iovsearch'] = subresults
         elif section == 'residual':
             func = partial(_run_resmod, path=db.path)
-            run_subfuncs['resmod'] = func
+            subresults = func(next_model)
+            next_model = subresults.best_model
+            run_subtools['resmod'] = subresults
         elif section == 'allometry':
             func = partial(_run_allometry, allometric_variable=allometric_variable, path=db.path)
-            run_subfuncs['allometry'] = func
+            subresults = func(next_model)
+            next_model = subresults.best_model
+            run_subtools['allometry'] = subresults
         elif section == 'covariates':
             func = partial(
                 _run_covariates, continuous=continuous, categorical=categorical, path=db.path
             )
-            run_subfuncs['covsearch'] = func
+            subresults = func(next_model)
+            next_model = subresults.best_model
+            run_subtools['covsearch'] = subresults
         else:
             raise ValueError(
                 f"Unrecognized section {section} in order. Must be one of {default_order}"
             )
```

Comment:
This will not allow to store subresults if a tool is present multiple times in `order`. Should use a list instead of a dictionary?